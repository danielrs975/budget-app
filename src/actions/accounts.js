/**
 * This script contain all the actions about the
 * accounts information
 * @author Daniel Rodriguez
 */
import uuid from 'uuid';

/**
 * This action add an account to the database
 * @param {*} account all the info about the account
 */
export const addAccount = (account) => ({
	type    : 'ADD_ACCOUNT',
	account
});

export const startAddAccount = (account) => {
	return (dispatch, getState) => {
		dispatch(addAccount({ id: uuid(), ...account }));
		const accounts = getState().accounts;
		localStorage.setItem('accounts', JSON.stringify(accounts));
	};
};

/**
 * This action is set in the state the accounts that are saved
 * in the db
 * @param {*} accounts The accounts to set in the state
 */
export const setAccount = (accounts) => ({
	type     : 'SET_ACCOUNTS',
	accounts
});

export const startSetAccount = () => {
	return (dispatch) => {
		const accounts = JSON.parse(localStorage.getItem('accounts'));
		dispatch(setAccount(accounts || []));
	};
};
