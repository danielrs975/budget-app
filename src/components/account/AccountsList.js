import React from 'react';
import AccountCard from './AccountCard';
import { Link } from 'react-router-dom';

const AccountsList = ({ accounts }) => (
	<div>
		{accounts.length > 0 &&
			accounts.map((account) => <AccountCard key={account.id} {...account} />)}
		<Link to="account/create">Add Account</Link>
	</div>
);

export default AccountsList;
