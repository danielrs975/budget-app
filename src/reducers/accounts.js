/**
 * This is the reducer for the accounts
 * information
 * @author Daniel Rodriguez
 */

export default (state = [], action) => {
	switch (action.type) {
		case 'ADD_ACCOUNT':
			return [ ...state, action.account ];
		default:
			return state;
	}
};
