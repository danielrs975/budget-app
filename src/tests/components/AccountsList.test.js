import React from 'react';
import { shallow } from 'enzyme';
import AccountsList from '../../components/AccountsList';
import accounts from '../fixtures/accounts';

test('should render AccountsList correctly', () => {
	const wrapper = shallow(<AccountsList accounts={accounts} />);
	expect(wrapper).toMatchSnapshot();
});
