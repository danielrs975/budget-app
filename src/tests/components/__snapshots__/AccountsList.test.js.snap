// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AccountsList correctly 1`] = `
<div>
  <AccountCard
    amount={1000}
    key="1"
    name="Revolut"
  />
  <AccountCard
    amount={1500}
    key="2"
    name="Societe Generale"
  />
  <AccountCard
    amount={5400}
    key="3"
    name="Cash"
  />
  <[object Object]
    to="account/create"
  >
    Add Account
  </[object Object]>
</div>
`;
